### PR TITLE
feat(lambda-python): support build args (#12949)

### LINKED ISSUE #12949: lambda-python: Allow passing build args to Docker in PythonFunction
Ability to pass args to docker when using `aws_lambda_python.PythonFunction`

### Use Case
I am trying to deploy a stack from behind a proxy. I want to be able to pass `--network host` to the PythonLambda docker command so that the container is able to access the internet through the proxy.

### Proposed Solution
This seems to have been solved for `aws_lambda_nodejs` here:
https://github.com/aws/aws-cdk/issues/8117

So perhaps a similar solution?

### Other
* [ ] :wave: I may be able to implement this feature request
* [ ] :warning: This feature might incur a breaking change

---

This is a :rocket: Feature Request

### PR DESCRIPTION
**Pull request by rirze**
Add support for passing build arguments when building the bundling
image.

Closes #12949

----

*By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license*

### CODE REVIEW

**Comment by rirze** [comment]
If possible, I'd like some help in adding support for passing in Docker environment variables for both `PythonFunction` and `PythonLayerVersion`. I tried taking a look at how it was implemented in `lambda-nodejs` but my Typescript comprehension wasn't strong enough to port over that functionality to this module.

File: packages/@aws-cdk/aws-lambda-python/README.md [hunk 1]
```diff
@@ -86,3 +89,15 @@ new lambda.PythonFunction(this, 'MyFunction', {
   ],
 });
 ```
+
+**Lambda with custom Docker Build Time Variables**
+
+Use the `build_args` parameter to pass build arguments when building the bundling image:
```
Comment (eladb) [inline packages/@aws-cdk/aws-lambda-python/README.md:95]:
```suggestion
Use the `buildArgs` parameter to pass build arguments when building the bundling image:
```

File: packages/@aws-cdk/aws-lambda-python/lib/bundling.ts [hunk 2]
```diff
@@ -71,6 +71,13 @@ export interface BundlingOptions {
    * @default - based on `assetHashType`
    */
   readonly assetHash?: string;
+
+  /**
+   * Build arguments to pass when building the bundling image.
+   *
+   * @default - no build arguments are passed
+   */
+  readonly buildArgs?: { [key:string] : string };
```
Comment (eladb) [inline packages/@aws-cdk/aws-lambda-python/lib/bundling.ts:80]: Stylistic:

```suggestion
  readonly buildArgs?: { [key:string]: string };
```

**Review by eladb — CHANGES_REQUESTED** [review]
Did you consider exposing a `bundling` option (like in `NodeJsFunction`) which already allows specifying `buildArgs`.

**Comment by rirze** [comment]
I did consider creating a bundling options class, but after looking at its implementation in `aws-lambda-nodejs/bundling.ts`, I couldn't separate what was essential to Docker and what was specific to Typescript bundling (esbuild). 

If desired, I could try my hand at exposing a `Bundling` class for this module and we could iteratively refine it. I'd imagine this would require a few rounds of feedback.

**Comment by eladb** [comment]
@rirze there is already a `BundlingOptions` struct, why not reuse that? (like `NodeJsFunction` does)

@jogold any chance you can work with @rirze to align those APIs?

**Comment by rirze** [comment]
Hmm, I think I was confused because I assumed `PythonFunction` would need its own `Bundling` class like it does in `NodeJsFunction` (`Bundling implements cdk.BundlingOptions`). But looking it more, it seems like most of the logic there is related to esbuild. So making sure I understand correctly, passing in a `cdk.BundlingOptions` should be sufficient for `PythonFunction`/`PythonLayerVersion`, correct?

**Comment by jogold** [comment]
We currently have the following API for the `NodejsFunction`:
* a `bundling` prop in `NodejsFunctionProps` of type `BundlingOptions`: https://github.com/aws/aws-cdk/blob/26919865f3f9079cbfa3103a4df14051f45fb734/packages/%40aws-cdk/aws-lambda-nodejs/lib/function.ts#L69-L75
* in there we have `buildArgs`: https://github.com/aws/aws-cdk/blob/26919865f3f9079cbfa3103a4df14051f45fb734/packages/%40aws-cdk/aws-lambda-nodejs/lib/types.ts#L175-L180

I think we can do the same here.

**Comment by eladb** [comment]
@rirze I am closing this PR for now. Feel free to re-open when ready to continue :-)

**Pull request closed without merging**